Bun doesn't run prisma generate or prisma migrate inside docker containers or WSL

### What version of Bun is running?

1.0.1+31aec4ebe325982fc0ef27498984b0ad9969162b

### What platform is your computer?

Microsoft Windows NT 10.0.22621.0 x64

### What steps can reproduce the bug?

# Docker:
```dockerfile
FROM oven/bun AS builder

WORKDIR /tmp/app

COPY package.json .
COPY bun.lockb .

RUN bun install

COPY src ./src
COPY tsconfig.json .

# This would be the optimal solution but it doesn't work, so I'm gonna use a hack here until I figure out how to fix it
#   RUN bunx prisma:generate
#   COPY prisma/schema.prisma ./prisma/schema.prisma
COPY node_modules/.prisma ./node_modules/.prisma

RUN bun run build

FROM oven/bun AS runner

ARG DEFAULT_DATABASE_URL= "file:./deploy.db"
ENV DATABASE_URL=$DEFAULT_DATABASE_URL

WORKDIR /app

COPY --from=builder /tmp/app/package.json /app/package.json
COPY --from=builder /tmp/app/bun.lockb /app/bun.lockb
RUN bun install -p

# Refer to the comment above
#   COPY --from=builder /tmp/app/prisma/schema.prisma ./prisma/schema.prisma
#   RUN bun run prisma:generate
COPY --from=builder /tmp/app/node_modules/.prisma /app/node_modules/.prisma

#   These commands aren't working either. The prisma:migrate has to be done in the external machine
#   RUN bun add prisma
#   RUN bun run prisma:migrate

COPY --from=builder /tmp/app/build /app/build

CMD ["bun", "start"]
```

# WSL:

Install WSL normally...
```powershell
wsl --install
```

Make bun installation inside Ubuntu.

Try to follow this exact guide: https://bun.sh/guides/ecosystem/prisma


### What is the expected behavior?

Prisma installs normally

### What do you see instead?

This, download never finishes:
```bash
root@7fa00e64baa1:/app# bun run prisma:deploy
$ bunx prisma migrate deploy
Downloading Prisma engines for Node-API for debian-openssl-1.1.x [===                 ] 15%root@7fa00e64baa1:/app#
```



### Additional information

_No response_

In Docker it fails silently and in local it throws this error:
```
✘ [ERROR] Could not resolve ".prisma/client/index-browser"

    node_modules/@prisma/client/index-browser.js:1:23:
      1 │ const prisma = require('.prisma/client/index-browser')
        ╵                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path ".prisma/client/index-browser" as external to
  exclude it from the bundle, which will remove this error. You can also
  surround this "require" call with a try/catch block to handle this
  failure at run-time instead of bundle-time.
```

Hi people.

Does somebody have this problem when using docker with prisma and bun?:

```
 error TS2305: Module '"@prisma/client"' has no exported member 'PrismaClient'.
 ```
 
If a runnint locally, this problem doesn't show up! bun running `docker build`, it fails!

> Hi people.
> Does somebody have this problem when using docker with prisma and bun?:
> ```
>  error TS2305: Module '"@prisma/client"' has no exported member 'PrismaClient'.
> ```
> If a run it locally, this problem doesn't show up! but running `docker build`, it fails!

This is exactly what I have! I'm running ReviOS 11 so I thought it's due to some of it's optimizations, I can never get `PrismaClient` to be detected and neither can i run any `bunx prisma` codes.... only `--help` works

--- a/src/bun.js/bindings/JSReadableHelper.cpp
+++ b/src/bun.js/bindings/JSReadableHelper.cpp
@@ -175,6 +175,7 @@ EncodedJSValue emitReadable_(JSGlobalObject* lexicalGlobalObject, JSObject* stre
     VM& vm = lexicalGlobalObject->vm();
     auto throwScope = DECLARE_THROW_SCOPE(vm);
     JSValue errored = state->m_errored.get();
+
     if (!state->getBool(JSReadableState::destroyed) && !errored.toBoolean(lexicalGlobalObject) && (state->m_length || state->getBool(JSReadableState::ended))) {
         // stream.emit('readable')
         auto clientData = WebCore::clientData(vm);
@@ -201,61 +202,7 @@ JSC_DEFINE_HOST_FUNCTION(jsReadable_emitReadable_, (JSGlobalObject * lexicalGlob
 {
     JSReadableHelper_EXTRACT_STREAM_STATE
 
-        emitReadable_(lexicalGlobalObject, stream, state);
-
-    RELEASE_AND_RETURN(throwScope, JSValue::encode(jsUndefined()));
-}
-
-EncodedJSValue emitReadable(JSGlobalObject* lexicalGlobalObject, JSObject* stream, JSReadableState* state)
-{
-    VM& vm = lexicalGlobalObject->vm();
-
-    state->setBool(JSReadableState::needReadable, false);
-    if (!state->getBool(JSReadableState::emittedReadable)) {
-        state->setBool(JSReadableState::emittedReadable, true);
-        Zig::GlobalObject* globalObject = reinterpret_cast<Zig::GlobalObject*>(lexicalGlobalObject);
-        globalObject->queueMicrotask(JSValue(globalObject->emitReadableNextTickFunction()), JSValue(stream), JSValue(state), JSValue {}, JSValue {});
-    }
-    return JSValue::encode(jsUndefined());
-}
-
-JSC_DEFINE_HOST_FUNCTION(jsReadable_emitReadable, (JSGlobalObject * lexicalGlobalObject, CallFrame* callFrame))
-{
-    JSReadableHelper_EXTRACT_STREAM_STATE
-
-        RELEASE_AND_RETURN(throwScope, emitReadable(lexicalGlobalObject, stream, state));
-}
-
-JSC_DEFINE_HOST_FUNCTION(jsReadable_onEofChunk, (JSGlobalObject * lexicalGlobalObject, CallFrame* callFrame))
-{
-    JSReadableHelper_EXTRACT_STREAM_STATE
-
-        if (state->getBool(JSReadableState::ended))
-            RELEASE_AND_RETURN(throwScope, JSValue::encode(jsUndefined()));
-
-    auto decoder = jsDynamicCast<JSStringDecoder*>(state->m_decoder.get());
-    if (decoder) {
-        JSString* chunk = jsDynamicCast<JSString*>(decoder->end(vm, lexicalGlobalObject, nullptr, 0));
-        if (chunk && chunk->length()) {
-            auto buffer = jsDynamicCast<JSBufferList*>(state->m_buffer.get());
-            if (!buffer) {
-                throwTypeError(lexicalGlobalObject, throwScope, "Not buffer on stream"_s);
-                return JSValue::encode(jsUndefined());
-            }
-            buffer->push(vm, JSValue(chunk));
-            state->m_length += state->getBool(JSReadableState::objectMode) ? 1 : chunk->length();
-        }
-    }
-
-    state->setBool(JSReadableState::ended, true);
-
-    if (state->getBool(JSReadableState::sync)) {
-        RELEASE_AND_RETURN(throwScope, emitReadable(lexicalGlobalObject, stream, state));
-    } else {
-        state->setBool(JSReadableState::needReadable, false);
-        state->setBool(JSReadableState::emittedReadable, true);
         RELEASE_AND_RETURN(throwScope, emitReadable_(lexicalGlobalObject, stream, state));
-    }
 }
 
 #undef JSReadableHelper_EXTRACT_STREAM_STATE

--- a/src/bun.js/bindings/JSReadableHelper.h
+++ b/src/bun.js/bindings/JSReadableHelper.h
@@ -6,8 +6,6 @@ namespace WebCore {
 
 JSC_DECLARE_HOST_FUNCTION(jsReadable_maybeReadMore);
 JSC_DECLARE_HOST_FUNCTION(jsReadable_resume);
-JSC_DECLARE_HOST_FUNCTION(jsReadable_emitReadable);
-JSC_DECLARE_HOST_FUNCTION(jsReadable_onEofChunk);
 JSC_DECLARE_HOST_FUNCTION(jsReadable_emitReadable_);
 
 } // namespace WebCore

--- a/src/bun.js/bindings/ZigGlobalObject.cpp
+++ b/src/bun.js/bindings/ZigGlobalObject.cpp
@@ -1913,10 +1913,7 @@ JSC_DEFINE_HOST_FUNCTION(functionLazyLoad,
                 JSC::JSFunction::create(vm, globalObject, 0, "resume"_s, jsReadable_resume, ImplementationVisibility::Public), 0);
             obj->putDirect(
                 vm, JSC::PropertyName(JSC::Identifier::fromString(vm, "emitReadable"_s)),
-                JSC::JSFunction::create(vm, globalObject, 0, "emitReadable"_s, jsReadable_emitReadable, ImplementationVisibility::Public), 0);
-            obj->putDirect(
-                vm, JSC::PropertyName(JSC::Identifier::fromString(vm, "onEofChunk"_s)),
-                JSC::JSFunction::create(vm, globalObject, 0, "onEofChunk"_s, jsReadable_onEofChunk, ImplementationVisibility::Public), 0);
+                JSC::JSFunction::create(vm, globalObject, 0, "emitReadable"_s, jsReadable_emitReadable_, ImplementationVisibility::Public), 0);
             return JSValue::encode(obj);
         }
         if (string == "events"_s) {

--- a/src/js/node/stream.js
+++ b/src/js/node/stream.js
@@ -2508,14 +2508,17 @@ var require_readable = __commonJS({
 
     var { addAbortSignal } = require_add_abort_signal();
     var eos = require_end_of_stream();
-    const { maybeReadMore: _maybeReadMore, resume, emitReadable: _emitReadable, onEofChunk } = $lazy("bun:stream");
+    const { maybeReadMore: _maybeReadMore, resume, emitReadable: _emitReadable } = $lazy("bun:stream");
     function maybeReadMore(stream, state) {
       process.nextTick(_maybeReadMore, stream, state);
     }
-    // REVERT ME
     function emitReadable(stream, state) {
       $debug("NativeReadable - emitReadable", stream.__id);
-      _emitReadable(stream, state);
+      state.needReadable = false;
+      if (!state.emittedReadable) {
+        state.emittedReadable = true;
+        process.nextTick(_emitReadable, stream, state);
+      }
     }
     var destroyImpl = require_destroy();
     var {
@@ -2623,6 +2626,24 @@ var require_readable = __commonJS({
       }
       $debug("about to maybereadmore");
       maybeReadMore(stream, state);
+    }
+    function onEofChunk(stream, state) {
+      if (state.ended) return;
+      if (state.decoder) {
+        const chunk = state.decoder.end();
+        if (chunk && chunk.length) {
+          state.buffer.push(chunk);
+          state.length += state.objectMode ? 1 : chunk.length;
+        }
+      }
+      state.ended = true;
+      if (state.sync) {
+        emitReadable(stream, state);
+      } else {
+        state.needReadable = false;
+        state.emittedReadable = true;
+        _emitReadable(stream, state);
+      }
     }
     Readable.prototype.isPaused = function () {
       const state = this._readableState;

--- a/src/js/node/stream.js
+++ b/src/js/node/stream.js
@@ -2629,11 +2629,14 @@ var require_readable = __commonJS({
     }
     function onEofChunk(stream, state) {
       if (state.ended) return;
-      if (state.decoder) {
-        const chunk = state.decoder.end();
-        if (chunk && chunk.length) {
+
+      const decoder = state.decoder;
+      if (decoder) {
+        const chunk = decoder.end();
+        const chunkLength = chunk?.length;
+        if (chunkLength) {
           state.buffer.push(chunk);
-          state.length += state.objectMode ? 1 : chunk.length;
+          state.length += state.objectMode ? 1 : chunkLength;
         }
       }
       state.ended = true;

--- a/src/js/private.d.ts
+++ b/src/js/private.d.ts
@@ -163,7 +163,6 @@ interface BunLazyModules {
     maybeReadMore: Function;
     resume: Function;
     emitReadable: Function;
-    onEofChunk: Function;
     ReadableState: Function;
   };
   sqlite: any;

--- a/test/js/node/stream/node-stream.test.js
+++ b/test/js/node/stream/node-stream.test.js
@@ -184,6 +184,20 @@ describe("createReadStream", () => {
       done();
     });
   });
+
+  it("should emit readable on end", done => {
+    const testData = "Hello world";
+    const path = join(tmpdir(), `${Date.now()}-testEmitReadableOnEnd.txt`);
+    writeFileSync(path, testData);
+    const stream = createReadStream(path);
+
+    stream.on("readable", () => {
+      const chunk = stream.read();
+      if (!chunk) {
+        done();
+      }
+    });
+  });
 });
 
 describe("Duplex", () => {

